tests/TestCase/TestSuite/Schema/SchemaCleanerTest.php

```diff
@@ -0,0 +1,132 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @since         4.3.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Test\TestCase\TestSuite\Schema;
+
+use Cake\Database\Schema\TableSchema;
+use Cake\Datasource\ConnectionManager;
+use Cake\TestSuite\Schema\SchemaCleaner;
+use Cake\TestSuite\TestCase;
+
+class SchemaCleanerTest extends TestCase
+{
+    public function testForeignKeyConstruction()
+    {
+        $connection = ConnectionManager::get('test');
+
+        $this->createSchemas();
+        $this->assertSame(
+            1,
+            $connection->newQuery()->select('id')->from('test_table')->execute()->count()
+        );
+        $this->assertSame(
+            1,
+            $connection->newQuery()->select('id')->from('test_table2')->execute()->count()
+        );
+
+        $this->expectException(\PDOException::class);
+        $connection->delete('test_table');
+    }
+
+    public function testDropSchema()
```

Comment:
The tests for `SchemaCleaner` are the ones causing tests to crash in postgres. Dropping all the tables in the connection results in an uncaught exception when the `FixtureManager` attempts to clean up fixtures. 

Some potential solutions I can think of are:

1. Test this method with a mocked cleaner that doesn't actually execute SQL.
2. Add parameters to only drop a subset of tables. If that parameter is omitted `$schema->listTables()` can be used.

I'm generally not a fan of mocks but this might be a good fit for them given the destructive nature of dropping tables.

Reply:
Thanks for the hint. I'll give option 2 a try.

Reply:
@markstory Did some refactoring to pass a set of tables when truncating/refactoring, in the line of the second solution you suggested. Let's see how the CI reacts to that...